jax/_src/dispatch.py

```diff
@@ -326,35 +326,37 @@ def not_none_device_or_backend_on_jit(backend, device, num_ins):
 
 
 def sharded_lowering(fun, device, backend, name, donated_invars, always_lower,
-                     keep_unused, *arg_specs):
+                     keep_unused, *arg_specs,
+                     lowering_platform: Optional[str]):
```

Comment:
Can you please move this before *arg_specs?

Reply:
It is after `*arg_specs` for a very precise reason: I want it to be kwarg. With that, and without a default value, you get a very clear error at a callsite where you forgot to set it that the function expects a `lowering_platform` argument. If we add it before `*arg_specs` then it will get confused with an argument and the error will be a lot more confusing. This has been very helpful when making this change and fixing all the places that need to be fixed. It is helpful also when people use these internal apis from other projects (e.g., search for `_pjit_lower(` in Google code base). Think also when we will remove this argument later on, if we make it positional and somebody forgets to remove it at the call site (e.g., these other projects) it will get gobbled as an argument, instead of getting a nice error message that the function got an unexpected `lowering_platform` argument. Do you agree?

Reply:
Ohh.. good point! Let's keep it then! Maybe add a small comment that this is supposed to be a kwarg?